rbi/annotations/railties.rbi

```diff
@@ -42,8 +42,16 @@ class Rails::Application < ::Rails::Engine
 end
 
 class Rails::Engine < ::Rails::Railtie
-  sig { params(block: T.untyped).returns(ActionDispatch::Routing::RouteSet) }
+  sig { params(block: T.nilable(T.proc.bind(ActionDispatch::Routing::Mapper).void)).returns(ActionDispatch::Routing::RouteSet) }
   def routes(&block); end
+
+  class << self
+    sig { returns(Pathname) }
+    def root; end
```

Comment:
Where is this defined? I can't see any `Rails::Engine.root` method.

Reply:
Looks like it's delegated https://github.com/rails/rails/blob/main/railties/lib/rails/engine.rb?ref=writesoftwarewell.com#L436

Reply:
`method_missing` 🤮 

https://github.com/rails/rails/blob/747f85f200e7bb2c1a31b4e26e5a5655e2dc0cdc/railties/lib/rails/railtie.rb#L221-L229

Reply:
> Looks like it's delegated https://github.com/rails/rails/blob/main/railties/lib/rails/engine.rb?#L436

That's `Rails::Engine#root` though, not the one I am asking about.

Reply:
> `method_missing` 🤮
> 
> https://github.com/rails/rails/blob/747f85f200e7bb2c1a31b4e26e5a5655e2dc0cdc/railties/lib/rails/railtie.rb#L221-L229

Right, but also that means all class methods are delegated to instance methods, not just `root`.

Reply:
Yeah I didn't get that far, `root` was the only sig I needed :D